File: 완전탐색/Python/최고의 집합.py
def solution(n, s):
    # n이 s보다 크면 합을 s로 만들 수 없음
    if n > s:
        return [-1]
    
    base = s // n
    remainder = s % n
    
    # base가 0인 경우, n <= s가 아니라면 이미 위에서 -1을 반환
    # 그렇지 않으면 base >=1 이므로 문제 없음
    
    # 최적의 집합: 나머지를 먼저 분배하여 높은 숫자부터 채움
    result = [base] * (n - remainder) + [base + 1] * remainder
    
    return result

# 테스트 케이스
print(solution(2, 9))  # 기대 출력: [5, 4]
print(solution(2, 1))  # 기대 출력: [-1]
print(solution(2, 8))  # 기대 출력: [4, 4]
print(solution(3, 8))  # 기대 출력: [3, 3, 2]
print(solution(5, 8))  # 기대 출력: [2, 2, 2, 1, 1]

File: 완전탐색/Java/최고의 집합.java
package 완전탐색.Java;

import java.util.Arrays;

public class 최고의 집합 {
    public int[] solution(int n, int s) {
        // n이 s보다 클 경우, 합을 s로 만들 수 없으므로 [-1] 반환
        if (n > s) {
            return new int[]{-1};
        }
        
        int[] result = new int[n];
            
        // 기본값과 나머지를 계산
        int base = s / n;
        int remainder = s % n;
                
        // 먼저 나머지 부분에 base + 1을 할당하여 높은 숫자부터 채움
        for (int i = 0; i < remainder; i++) {
            result[i] = base + 1;
        }
            
        // 나머지 요소에 base를  할당
        for (int i = remainder; i < n; i++) {
            result[i] = base;
        }
            
        // 마지막으로 오름차 순 정렬
        Arrays.sort(result);
        return result;
    }
}
